src/routeGeneration/templates/express.ts

```diff
@@ -98,18 +98,21 @@ export function RegisterRoutes(app: any) {
     }
     {{/if}}
 
+    function isController(object: any): object is Controller {
+        return 'getHeaders' in object && 'getStatus' in object && 'setStatus' in object;
+    }
+
     function promiseHandler(controllerObj: any, promise: any, response: any, next: any) {
         return Promise.resolve(promise)
             .then((data: any) => {
                 let statusCode;
-                if (controllerObj instanceof Controller) {
-                    const controller = controllerObj as Controller
-                    const headers = controller.getHeaders();
+                if (isController(controllerObj)) {
```

Comment:
I think it might be clearer to just inline these checks (e.g. see if controllerObj.getHeaders exists and is a function, and then call it) rather than extracting the function

Reply:
Actually, this is fine.